src/documents/data_models.py

```diff
@@ -0,0 +1,125 @@
+import dataclasses
+import datetime
+import enum
+import hashlib
+from pathlib import Path
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import magic
+from dateutil.parser import isoparse
+
+
+@dataclasses.dataclass
+class DocumentOverrides:
+    """
+    Manages overrides for document fields which normally would
+    be set from content or matching.  All fields default to None,
+    meaning no override is happening
+    """
+
+    filename: Optional[str] = None
+    title: Optional[str] = None
+    correspondent_id: Optional[int] = None
+    document_type_id: Optional[int] = None
+    tag_ids: Optional[List[int]] = None
+    created: Optional[datetime.datetime] = None
+    asn: Optional[int] = None
+    owner_id: Optional[int] = None
+
+    def as_dict(self) -> Dict:
+        return {
+            "filename": self.filename,
+            "title": self.title,
+            "correspondent_id": self.correspondent_id,
+            "document_type_id": self.document_type_id,
+            "archive_serial_num": self.asn,
+            "tag_ids": self.tag_ids,
+            "created": self.created.isoformat() if self.created else None,
+            "owner_id": self.owner_id,
+        }
+
+    @staticmethod
+    def from_dict(data: Optional[Dict]) -> "DocumentOverrides":
+        if data is None:
+            return DocumentOverrides()
+        return DocumentOverrides(
+            data["filename"],
+            data["title"],
+            data["correspondent_id"],
+            data["document_type_id"],
+            data["tag_ids"],
+            isoparse(data["created"]) if data["created"] else None,
+            data["archive_serial_num"],
+            data["owner_id"],
+        )
+
+
+class DocumentSource(enum.IntEnum):
```

Comment:
Definitely could see this data being useful in classification logic down the line